Gems/RecastNavigation/Code/Include/RecastNavigation/RecastNavigationMeshBus.h

```diff
@@ -17,13 +17,68 @@ namespace RecastNavigation
 {
     //! Holds a pointer to Recast navigation mesh and the associated mutex.
     //! This structure is to be used when performing operations on a navigation mesh.
-    struct NavMeshQuery
+    //! In order to access NavMesh or NavMeshQuery objects, use the object returned by @AcquireLock.
+    class NavMeshQuery
     {
+    public:
+        class LockGuard;
+
+        NavMeshQuery(dtNavMesh* navMesh, dtNavMeshQuery* navQuery)
+        {
+            m_mesh.reset(navMesh);
+            m_query.reset(navQuery);
+        }
+
+        //! In order to access the underlying navigation objects, acquire a lock.
+        LockGuard AcquireLock()
```

Comment:
Be careful with this.. without RTO this will acquire, release, and then acquire again on function return..

Reply:
Heh, I have been worrying about the same this today. For this reason, I already added `AZ_DISABLE_COPY_MOVE(LockGuard);` in hopes of preventing that and thus essentially forcing RTO.

Reply:
I removed AcquireLock.